packages/libs/src/sdk/utils/parseMintToken.ts

```diff
@@ -0,0 +1,31 @@
+import { PublicKey } from '@solana/web3.js'
+
+import { TokenName } from '../services'
+
+/**
+ * Helper function to get both the token and mint address from one or the other.
+ * @param mintOrToken either the name of the token ('USDC') or its mint address
+ * @param mints a mapping of tokens to mint addresses
+ * @returns the mint address and token name
+ * @throws if the token isn't in the mint map
+ */
+export const parseMintToken = (
+  mintOrToken: PublicKey | TokenName,
+  mints: Partial<Record<TokenName, PublicKey>>
+) => {
+  const mint =
+    mintOrToken instanceof PublicKey ? mintOrToken : mints[mintOrToken]
```

Comment:
`instanceof` still safe?

Reply:
Good callout... haven't tested this on mobile where I think these problems are. Not sure what to do here... maybe a try-catch around initializing a public key? hmm

Reply:
Going to leave for now as this isn't changing existing behavior anywhere... the `PublicKeySchema` in sdk still uses instanceof